plugins/rich-editor/src/scripts/components/EditorEmojiMenu.jsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * @author Stéphane (slafleche) LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+import React from "react";
+import * as PropTypes from "prop-types";
+import Quill from "quill/quill";
+import { t } from "@core/utility";
+// import Events from "@core/events";
+// import EditorToolbar from "./EditorToolbar";
+// import Emitter from "quill/core/emitter";
+// import { Range } from "quill/core/selection";
+import EditorEmojiButton from "../components/EditorEmojiButton";
+import emojis from 'emojibase-data/en/data.json';
+
+export default class EditorEmojiMenu extends React.Component {
+
+
+    static propTypes = {
+        quill: PropTypes.instanceOf(Quill).isRequired,
+        menuTitleID: PropTypes.string.isRequired,
+        isVisible: PropTypes.bool.isRequired,
+        closeMenu: PropTypes.func.isRequired,
+    };
+
+    /**
+     * @inheritDoc
+     */
+    constructor(props) {
+        super(props);
+
+        // Quill can directly on the class as it won't ever change in a single instance.
+        this.quill = props.quill;
+        this.menuTitleID = props.menuTitleID;
+        this.emojiList = emojis;
+        this.closeMenu = props.closeMenu;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    render() {
+        if(this.props.isVisible) {
```

Comment:
There seems to be a delay when you click the menu because it is essentially re-rendering it from scratch every time (although some elements may be cached). It would be more performant to render immediately once and just toggle display/visibility properties with CSS. In react a partial change is a lot more performant than fully re-rendering 1000+ DOM nodes.

Reply:
I'll make a separate issue for optimizing the emojis